modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -705,11 +705,36 @@ class CV_EXPORTS_W QRCodeDetector
     /** @brief Both detects and decodes QR code
 
      @param img grayscale or color (BGR) image containing QR code.
-     @param points opiotnal output array of vertices of the found QR code quadrangle. Will be empty if not found.
+     @param points optional output array of vertices of the found QR code quadrangle. Will be empty if not found.
      @param straight_qrcode The optional output image containing rectified and binarized QR code
      */
     CV_WRAP cv::String detectAndDecode(InputArray img, OutputArray points=noArray(),
                                         OutputArray straight_qrcode = noArray());
+    /** @brief Detects QR codes in image and returns the vector of the quadrangle containing the codes.
+     @param img grayscale or color (BGR) image containing (or not) QR codes.
+     @param points Output vector of vector of vertices of the minimum-area quadrangle containing the codes.
+     */
+    CV_WRAP bool detectMulti(InputArray img, OutputArrayOfArrays points) const;
```

Comment:
> OutputArrayOfArrays
> InputArrayOfArrays
> std::vector< std::vector< Point > > corners;

Please avoid using of `*OfArrays` types and/or nested `std::vector`.
This is unnecessary for fixed size elements (4 points per QR code).

Lets store `points` as one "array" (without nesting):
- `std::vector< Point >` where each 4 points define QR code instance (`<0, 1, 2, 3>, <4, 5, 6 ,7> ...`).
- `Mat` - CV_32FC2 with one or multiple rows.
and pass them as `OutputArray`.

This is continued discussion from this comment: https://github.com/opencv/opencv/pull/15338#discussion_r339004770